pkg/platform/platform.go

```diff
@@ -201,6 +203,9 @@ type Platform interface {
 	// GetDefaultRegistryCredentialsSecretName returns secret with credentials to push/pull from docker registry
 	GetDefaultRegistryCredentialsSecretName() string
 
+	// GetFunctionSecrets returns all the function's secrets
+	GetFunctionSecrets(ctx context.Context, functionName, functionNamespace string) ([]v1.Secret, error)
```

Comment:
👀 you are returning a list of kube secrets, would not be true for local platform (where secrets would be supported as well). Id suggest perhaps wrapping that up in an object that holds the type and each platorm will access that it needs. e.g.:

```go
GetFunctionSecrets(ctx context.Context, functionName, functionNamespace string) ([]functionSecret, error)
```

```go
type functionSecret struct {
  kubernetes *v1.Secret
  local *String
}
```


```go

// in kube-platform
functionSecrets, _ := GetFunctionSecrets(ctx, "my-func", "some-ns")
kubeClient.createSecret(functionSecret.kubernetes)
```